src/HexagonOptimize.cpp

```diff
@@ -178,7 +180,14 @@ Expr apply_patterns(Expr x, const vector<Pattern> &patterns, const Target &targe
     vector<Expr> matches;
     for (const Pattern &p : patterns) {
 
-        if ((p.flags & (Pattern::v62)) && !target.has_feature(Target::HVX_v62))
+        if ((p.flags & (Pattern::v62orLater)) &&
+            !target.features_any_of({Target::HVX_v62, Target::HVX_v65, Target::HVX_v66}))
```

Comment:
I'm not a big fan of how we are handling architecture versions in Target, but one of the reasons it is done this way is some architectures (e.g AVX) are no longer doing easily orderable supersets. I'm not sure I want to move to a style that puts extra flags in the Target.

Really I think we need to change our data structure to better represent the concept.